spectator/meter/max_gauge.go

```diff
@@ -0,0 +1,36 @@
+package meter
+
+import (
+	"fmt"
+	"github.com/Netflix/spectator-go/spectator/writer"
+)
+
+// MaxGauge represents a value that is sampled at a specific point in time. One
+// example might be the pending messages in a queue. This type is safe for
+// concurrent use.
+//
+// You can find more about this type by viewing the relevant Java Spectator
+// documentation here:
+//
+// https://netflix.github.io/spectator/en/latest/intro/gauge/
+type MaxGauge struct {
+	id              *Id
+	writer          writer.Writer
+	meterTypeSymbol rune
+}
+
+// NewMaxGauge generates a new gauge, using the provided meter identifier.
+func NewMaxGauge(id *Id, writer writer.Writer) *MaxGauge {
+	return &MaxGauge{id, writer, 'm'}
+}
+
+// MeterId returns the meter identifier.
+func (g *MaxGauge) MeterId() *Id {
+	return g.id
+}
+
+// Set records the current value.
+func (g *MaxGauge) Set(value float64) {
```

Comment:
Since we have `Add(int64)` and `AddFloat(float64)` methods on Counters, should we be following that pattern for these other meters, for consistency? That might be a moderate breaking change in the API though.

Reply:
I can add `Set` and `SetFloat`, but that'd mean breaking backwards compatibility

Reply:
Hm... let's keep it as-is for now to avoid breaking the API and have another think if a user asks about it.